### DISCUSSION TITLE
Add multiple attributes instead of a single attribute

### DISCUSSION THREAD

**Discussion by bkspace** [Shopify/function-examples#493] [API Ideas]
![CleanShot 2024-05-01 at 15 17 56@2x](https://github.com/Shopify/function-examples/assets/1695350/a504f695-94f5-4b01-af5e-630d160c83af)

Currently, on both the cart, and specific cart lines, we can access a specific property/attribute by passing a singular key (String). We can also do this for multiple using GQL aliases, and providing different keys. This can further, be made dynamic using input variables.

It would be a lovely update if we could add `attributes` (plural), e.g

```
pub struct Attribute {
  pub key: String,
  pub value: Option<String>,
}

pub type Attributes = Vec<Attribute>
```

Whereby input is: 
keys ( [String] ) | The keys of the attributes to check. Default value: []

**Comment by awd** [comment]
100% agree, as this would also allow pass-thru of attributes added by themes, other apps, etc.

Currently only specifying known attributes breaks across all types of integrations in ways that are quite confusing to merchants. 

Please re-prioritize this type of request - storefront integrations are currently stuck here.

I know there are opinions internally on how to handle unbounded lists, so maybe it is `attributes(limit: 25)`or something like this? 
In my observations I rarely see more than 25 attributes per-line when combined with scheduled delivery, gift options, personalization, warranty type of apps. 

Ideally this would apply to both cart and cart line attributes, and would allow a Cart Transform function to pull the attributes, modify [if needed] and pass them into the bundled or unbundled items.
